packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -139,6 +147,7 @@ const AbstractModal = (props) => {
         zIndex={zIndex}
         aria-modal="true"
         ref={modalElementRef}
+        onKeyDown={handleKeydown}
```

Comment:
How does splitting up the handlers solve this problem?

Reply:
The stopPropagation that I added in form-select was not stopping the document keydown event from bubbling to the select. So added keydown to the modal for which I was able to stop the bubbling using stopPropagation.